letsencrypt/default_server.nginx.jinja

```diff
@@ -0,0 +1,32 @@
+# default servers for nginx
+# this vHosts will catch all request that are not match an other vHosts server_name.
+server {
+    listen *:80    default_server;
+    listen [::]:80 default_server;
+    return 301 https://{{ fqdn }}$request_uri;
+}
+
+server {
+    listen *:443 default_server;
+    listen [::]:443 default_server;
+
+    ssl_protocols TLSv1.3;
+    ssl_session_cache 'shared:SSL:10m';
+    ssl_prefer_server_ciphers on;
+    ssl_stapling off;
+    ssl_stapling_verify off;
+    ssl_session_tickets off;
+
+    # Security headers
+    add_header X-Content-Type-Options $x_content_type_options;
+    add_header X-Frame-Options $x_frame_options;
+    add_header X-Robots-Tag $x_robots_tag;
+    add_header Referrer-Policy $referrer_policy;
+
+    # SSL certificate
+    ssl_dhparam {{ acme_certificate_dir }}/dhparam.pem;
+    ssl_certificate {{ acme_certificate_dir }}/{{ fqdn }}/fullchain.pem;
+    ssl_certificate_key {{ acme_certificate_dir }}/{{ fqdn }}/privkey.pem;
+
+    return 301 https://{{ fqdn }}$request_uri;
```

Comment:
I'm more inclined to not redirect but return an HTTP code that indicates something is not right with the request, instead of just redirecting. Depeding on the clients behavior, this might also result in a forward loop? (if the header is still not correct when requesting the redirected resource)

I think `400` (Bad Request) might be a good candiate?

Reply:
I changed it to `400` and add a status text.

The response will then look like this if you set the Host header to `google.de`. 
```
HTTP/2 400 
server: nginx
date: Wed, 25 Jan 2023 08:29:04 GMT
content-type: application/octet-stream
content-length: 23

Unknown Host google.de
```

I was also thinking about `404` as HTTP code. But I think `400` is also fine.